id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java

```diff
@@ -1,700 +1,709 @@
+	 */
+	private void updateCbeff(Uin uinObject, RequestDTO requestDTO,AnonymousProfileDto anonymousProfileDto) throws IdRepoAppException {
+		ObjectNode identityMap = (ObjectNode) convertToObject(uinObject.getUinData(), ObjectNode.class);
+
+		uinObject.getBiometrics().stream().forEach(bio -> requestDTO.getDocuments().stream()
+				.filter(doc -> StringUtils.equals(bio.getBiometricFileType(), doc.getCategory())).forEach(doc -> {
+					try {
+						String uinHash = uinObject.getUinHash().split("_")[1];
+						if (!objectStoreHelper.biometricObjectExists(uinHash, bio.getBioFileId())) {
+							mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_IMPL, "updateCbeff",
+									"FILE NOT FOUND IN OBJECT STORE");
+							throw new IdRepoAppUncheckedException(FILE_NOT_FOUND);
+						}
+						byte[] data = objectStoreHelper.getBiometricObject(uinHash, bio.getBioFileId());
+						if (StringUtils.equalsIgnoreCase(
+								identityMap.get(bio.getBiometricFileType()).get(FILE_FORMAT_ATTRIBUTE).asText(),
+								CBEFF_FORMAT) && bio.getBioFileId().endsWith(CBEFF_FORMAT)) {
+							byte[] decodedBioData = CryptoUtil.decodeBase64(doc.getValue());
+							anonymousProfileDto.setOldCbeff(CryptoUtil.encodeBase64(data));
+							doc.setValue(CryptoUtil.encodeBase64(this.updateXML(decodedBioData, data,anonymousProfileDto)));
+						}
+					} catch (Exception e) {
+						mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_IMPL, "updateCbeff",
+								"\n" + ExceptionUtils.getStackTrace(e));
+						throw new IdRepoAppUncheckedException(INVALID_INPUT_PARAMETER.getErrorCode(), String.format(
+								INVALID_INPUT_PARAMETER.getErrorMessage(), DOCUMENTS + " - " + doc.getCategory()));
+					}
+				}));
+	}
+	
+	private byte[] updateXML(byte[] inputBioData, byte[] existingBioData,AnonymousProfileDto anonymousProfileDto) throws Exception {
+		List<BIR> existingBIRData = CbeffValidator.convertBIRTypeToBIR(cbeffUtil.getBIRDataFromXML(existingBioData));
+		List<BIR> inputBIRData = CbeffValidator.convertBIRTypeToBIR(cbeffUtil.getBIRDataFromXML(inputBioData));
+		Map<String, BIR> inputBIRDataMap = inputBIRData.stream()
+				.collect(Collectors.toMap(bir -> bir.getBdbInfo().getType().stream().map(bioType -> bioType.value())
+						.collect(Collectors.joining())
+						.concat(bir.getBdbInfo().getSubtype().stream().collect(Collectors.joining())), bir -> bir));
+		Map<String, BIR> existingBIRDataMap = existingBIRData.stream()
+				.collect(Collectors.toMap(bir -> bir.getBdbInfo().getType().stream().map(bioType -> bioType.value())
+						.collect(Collectors.joining())
+						.concat(bir.getBdbInfo().getSubtype().stream().collect(Collectors.joining())), bir -> bir));
+		inputBIRDataMap.entrySet().forEach(entry -> existingBIRDataMap.replace(entry.getKey(), entry.getValue()));
+		byte[] updatedCbeff = cbeffUtil.createXML(new ArrayList<>(existingBIRDataMap.values()));
+		anonymousProfileDto.setNewCbeff(CryptoUtil.encodeBase64(updatedCbeff));
```

Comment:
Instead of passing anonymousProfileDto to this method, can we set the new cbeff after this method call ? so we avoid second time base 64 encoding of the cbeff.

Reply:
Now with the recent changes it will not encode second time

Reply:
Resolved